### PR TITLE
Feature/adf 305/hidden task support

### PR DESCRIPTION
**Pull request by boajer**
https://oat-sa.atlassian.net/browse/ADF-305

Dependent PRs:

- [ ] https://github.com/oat-sa/extension-tao-task-queue/pull/131
- [ ] https://github.com/oat-sa/extension-tao-advanced-search/pull/26

### CODE REVIEW

File: actions/class.TaskQueueWebApi.php [hunk 1]
```diff
@@ -36,54 +37,46 @@
...
+    use HttpJsonResponseTrait;
 
-    const PARAMETER_TASK_ID = 'taskId';
-    const PARAMETER_LIMIT = 'limit';
-    const PARAMETER_OFFSET = 'offset';
-    const ALL = 'all';
+    private const PARAMETER_TASK_ID = 'taskId';
+    private const PARAMETER_LIMIT = 'limit';
+    private const PARAMETER_OFFSET = 'offset';
+    private const ALL = 'all';
```
Comment (gabrielfs7) [inline actions/class.TaskQueueWebApi.php:49]: 🚀

File: models/classes/taskQueue/TaskLog/TaskLogFilter.php [hunk 2]
```diff
@@ -193,13 +152,8 @@ public function addFilter($field, $operator, $value)
...
     /**
      * Add a basic filter to query only rows belonging to a given user and not having status ARCHIVED or CANCELLED.
-     *
-     * @param string $userId
-     * @param bool $archivedAllowed
-     * @param bool $cancelledAvailable
-     * @return $this
      */
-    public function addAvailableFilters($userId, $archivedAllowed = false, $cancelledAvailable = false)
+    public function addAvailableFilters(string $userId, bool $archivedAllowed = false, bool $cancelledAvailable = false): self
```
Comment (gabrielfs7) [inline models/classes/taskQueue/TaskLog/TaskLogFilter.php:156]: Just double checking. Are we 100% the callers are respecting strict types?

File: models/classes/taskQueue/TaskLog/TaskLogFilter.php [hunk 3]
```diff
@@ -383,7 +284,12 @@ private function assertValidOperator($op)
         ];
 
         if (!in_array($op, $operators)) {
-            throw new \InvalidArgumentException('Operator "' . $op . '"" is not a valid operator.');
+            throw new InvalidArgumentException('Operator "' . $op . '"" is not a valid operator.');
         }
     }
+
+    private function getIgnoredTasks(): array
```
Comment (gabrielfs7) [inline models/classes/taskQueue/TaskLog/TaskLogFilter.php:291]: No need for a private getter IMO. Just use `$this->ignoredTasks` directly

File: models/classes/taskQueue/TaskLog/TaskLogFilter.php (hunk 2 above)
Comment (boajer) [reply models/classes/taskQueue/TaskLog/TaskLogFilter.php:156]: My research and briefe testing shown that we are safe

File: models/classes/taskQueue/Queue.php [hunk 4]
```diff
@@ -145,7 +145,7 @@ public function setBroker(QueueBrokerInterface $broker)
      *
      * @return QueueBrokerInterface
      */
-    protected function getBroker()
+    public function getBroker()
```
Comment (gabrielfs7) [inline models/classes/taskQueue/Queue.php:148]: Why is this public? I see you are not using it outside.

**Review by gabrielfs7 — APPROVED** [review]
Take a look in the comments, but in general looks good 

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: models/classes/taskQueue/Queue.php (hunk 4 above)
Comment (boajer) [reply models/classes/taskQueue/Queue.php:148]: https://github.com/oat-sa/extension-tao-advanced-search/pull/26/files#diff-8c189e9e937bcbb92186e08a0e2ed545919bd0997b4bb5a160e35ed7e3fa1ddfR85

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Pull request merged**